### PR TITLE
Feature/40931 signaling included properties of the projects api

### PR DESCRIPTION
**Pull request by ulferts**
Enables signaling by API clients so that only desired properties are returned. 

For example, only the `total` and `count` of a project `Collection`-resource as well as the `name`, `id` and `ancestors`-links of project resources, embedded as results can be returned. This would be done by calling

```
/api/v3/projects?select=total,count,embedded/name,embedded/id,embedded/ancestors
```

This relies on the SqlRepresenters we have introduced for capabilities and because of that, the functionality was also added for the capabilities endpoint. Only end points relying on the SqlRepresenters will support signaling and because of this, the functionality is still limited to only a few end points. 

The main focus of the PR is on the projects#index end point so that fetching projects for autocompleters is done quickly:
* https://community.openproject.org/projects/openproject/work_packages/40330 
* https://community.openproject.org/projects/openproject/work_packages/40931

However, other endpoints have also received stubs for sql based representers:
* /api/v3/users
* /api/v3/groups
* /api/v3/placeholder_users
* /api/v3/principals
* /api/v3/work_packages
* /api/v3/projects/{id}/work_packages

to see whether the concept is applicable and the solution works on a general basis.

# TODO

* [x] Enable signaling for the capabilities end point (done because it is easier to start with)
* [x] Create an SqlRepresenter for projects. This will probably not be possible to support everything since embedded resources would also have to be implemented based on the SqlRepresenter.
  * [x] `id`
  * [x] `name`
  * [x] `ancestors` 
* [x] Switch to using the SqlRepresenter in case signaling parameters are provided
* [x] Also support providing the select parameter as an array of strings as that would be more in line with the rest of the parameters
* [x] Error messages in case an unsupported signaling is provided

### CODE REVIEW

**Comment by ulferts** [comment]
I conducted a preliminary test with `ab` comparing signaled requests to unsignaled. Since the SQL based representer currently only has the `id` and the `name` property, the results can only hint at the performance to be expected once more properties are supported. So far however, the results look very promising.

### Non signaled (`representable`-rendered)

```
Completed 100 requests
Completed 200 requests
Finished 200 requests


Server Software:        
Server Hostname:        localhost
Server Port:            3000

Document Path:          /api/v3/projects?pageSize=1000
Document Length:        830206 bytes

Concurrency Level:      1
Time taken for tests:   443.752 seconds
Complete requests:      200
Failed requests:        0
Total transferred:      166293000 bytes
HTML transferred:       166041200 bytes
Requests per second:    0.45 [#/sec] (mean)
Time per request:       2218.758 [ms] (mean)
Time per request:       2218.758 [ms] (mean, across all concurrent requests)
Transfer rate:          365.96 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.0      0       0
Processing:  1812 2219 265.1   2208    2998
Waiting:     1812 2218 265.1   2207    2998
Total:       1812 2219 265.1   2208    2998

Percentage of the requests served within a certain time (ms)
  50%   2208
  66%   2342
  75%   2403
  80%   2465
  90%   2574
  95%   2694
  98%   2777
  99%   2954
 100%   2998 (longest request)

``` 

### Signaled (SQL-rendered)

```
Completed 100 requests
Completed 200 requests
Finished 200 requests


Server Software:        
Server Hostname:        localhost
Server Port:            3000

Document Path:          /api/v3/projects?pageSize=1000&select=elements/id,total,count,self,elements/name
Document Length:        31979 bytes

Concurrency Level:      1
Time taken for tests:   13.567 seconds
Complete requests:      200
Failed requests:        0
Total transferred:      6647600 bytes
HTML transferred:       6395800 bytes
Requests per second:    14.74 [#/sec] (mean)
Time per request:       67.834 [ms] (mean)
Time per request:       67.834 [ms] (mean, across all concurrent requests)
Transfer rate:          478.50 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.0      0       0
Processing:    57   68  17.7     63     213
Waiting:       57   67  17.5     63     209
Total:         57   68  17.7     63     214

Percentage of the requests served within a certain time (ms)
  50%     63
  66%     65
  75%     67
  80%     68
  90%     73
  95%     88
  98%    157
  99%    165
 100%    214 (longest request)
 ```

**Comment by ulferts** [comment]
The failing feature spec is unrelated. It is simply flickering which I hope to have fixed in 4c4f13a6635e8d5ab52e5d7da7c39b5c9496c559

File: docs/api/apiv3/tags/signaling.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+description: |-
+  Some endpoints, especially those returning `Collection` resources, support signaling desired properties. By signaling, the client
+  can convey to the server the properties to include in a response.
+
+  Currently only `select` is supported which allows to specify the subset of properties a client is interested in. The benefit of using `select`
+  is increased response time. Other signaling, especially expanding the embedded resources to include as well over multiple layers of embedding
```
Comment (cbliard) [inline docs/api/apiv3/tags/signaling.yml]:
```suggestion
  is reduced response time. Other signaling, especially expanding the embedded resources to include as well over multiple layers of embedding
```

**Review by cbliard — COMMENTED** [review]
a minor remark

**Review by oliverguenther — APPROVED** [review]
I couldn't find any issues in the code or with the responses in my tests. Great job :clap:

**Pull request merged**